gradle-plugin/build.gradle.kts

```diff
@@ -24,6 +24,7 @@ plugins {
 dependencies {
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin-api:$kotlinBaseVersion")
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinBaseVersion")
+    implementation("org.jetbrains.kotlin:kotlin-compiler-runner:$kotlinBaseVersion")
```

Comment:
KGP has `compileOnly` dependency on this library as all classes should come from `kotlin-compiler-embeddable` on runtime. Probably you should do the same